### PR TITLE
feat: managed worker pools

### PR DESCRIPTION
**Pull request by abelanger5**
# Description

Frontend for managed worker pools.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: frontend/app/src/components/cloud/logging/logs.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+              }}
+            />
+          </p>
+        );
+      })}
+    </div>
+  );
+};
+
+const generateHash = (input: string): string => {
```
Comment (grutt) [inline frontend/app/src/components/cloud/logging/logs.tsx:163]: why're we doing this fe?

Comment (grutt) [inline frontend/app/src/components/cloud/logging/logs.tsx:163]: Why are we doing this FE?

File: frontend/app/src/components/molecules/time-picker/time-picker-input.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+const TimePickerInput = React.forwardRef<
+  HTMLInputElement,
+  TimePickerInputProps
+>(
+  (
+    {
+      className,
+      type = 'tel',
```
Comment (grutt) [inline frontend/app/src/components/molecules/time-picker/time-picker-input.tsx]:
```suggestion
      type = 'datetime-local',
```

File: frontend/app/src/pages/main/index.tsx [hunk 3]
```diff
@@ -74,6 +74,27 @@ function Sidebar({ className, memberships, currTenant }: SidebarProps) {
...
+    <SidebarButtonSecondary
+      key={1}
+      onNavLinkClick={onNavLinkClick}
+      to="/workers/all"
+      name="All Workers"
+    />,
+  ];
+
+  if (meta?.data.canLinkGithub) {
+    workers.push(
```
Comment (grutt) [inline frontend/app/src/pages/main/index.tsx:89]: -- todo: what does this look like if only the 1 elm is in worker

File: frontend/app/src/pages/main/index.tsx [hunk 4]
```diff
@@ -153,6 +176,12 @@ function Sidebar({ className, memberships, currTenant }: SidebarProps) {
                   <SidebarButtonSecondary
                     key={3}
                     onNavLinkClick={onNavLinkClick}
+                    to="/tenant-settings/github"
+                    name="Github"
+                  />,
+                  <SidebarButtonSecondary
+                    key={2}
```
Comment (grutt) [inline frontend/app/src/pages/main/index.tsx]:
```suggestion
                    key={4}
```

File: frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-activity.tsx [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+function EventList({
+  managedWorker,
+  setBuildId,
+}: {
+  managedWorker: ManagedWorker | undefined;
+  setBuildId: (id: string) => void;
+}) {
+  const getLogsQuery = useQuery({
+    ...queries.cloud.listManagedWorkerEvents(managedWorker?.metadata.id || ''),
```
Comment (grutt) [inline frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-activity.tsx]:
```suggestion
    ...queries.cloud.listManagedWorkerEvents(managedWorker!.metadata.id),
```
enabled should cover unknown?

File: frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-activity.tsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+      <div className="flex flex-row justify-start gap-4 items-center">
+        <Button
+          onClick={back}
+          variant="link"
+          className="flex items-center gap-1"
+        >
+          <ChevronLeftIcon className="w-4 h-4" />
+          Back
+        </Button>
+        {/* <h3 className="text-lg font-semibold leading-tight text-foreground">
```
Comment (grutt) [inline frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-activity.tsx]: rm comment

File: frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-metrics.tsx [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+      queryParams,
+    ),
+    enabled: !!managedWorker,
+    refetchInterval: () => {
+      return 5000;
+    },
+  });
+
+  const getDiskMetricsQuery = useQuery({
+    ...queries.cloud.getManagedWorkerDiskMetrics(
```
Comment (grutt) [inline frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-metrics.tsx:57]: future: do we want to combine metrics into one query?

File: frontend/app/src/pages/main/workers/managed-workers/create/components/create-worker-form.tsx [hunk 8]
```diff
@@ -0,0 +1,598 @@
...
+import { z } from 'zod';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Label } from '@/components/ui/label';
+import { Alert, AlertDescription, AlertTitle } from '@/components/ui/alert';
+import { Input } from '@/components/ui/input';
+import { Step, Steps } from '@/components/ui/steps';
+import EnvGroupArray, { KeyValueType } from '@/components/ui/envvar';
+
+export const machineTypes = [
```
Comment (grutt) [inline frontend/app/src/pages/main/workers/managed-workers/create/components/create-worker-form.tsx:22]: put this server side?

File: frontend/app/src/pages/main/workers/managed-workers/index.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+export default function ManagedWorkers() {
+  const { tenant } = useOutletContext<TenantContextType>();
+  invariant(tenant);
+
+  return (
+    <div className="flex-grow h-full w-full">
+      <div className="mx-auto max-w-7xl py-8 px-4 sm:px-6 lg:px-8">
+        <div className="flex flex-row justify-between items-center">
+          <h2 className="text-2xl font-bold leading-tight text-foreground">
+            Managed Worker Pools
```
Comment (grutt) [inline frontend/app/src/pages/main/workers/managed-workers/index.tsx:17]: helper text?

**Review by grutt — APPROVED** [review]

File: frontend/app/src/components/cloud/logging/logs.tsx (hunk 1 above)
Comment (abelanger5) [reply frontend/app/src/components/cloud/logging/logs.tsx:163]: there are definitely better ways to generate a stable key but will tackle this once logging perf becomes an issue

File: frontend/app/src/pages/main/index.tsx (hunk 3 above)
Comment (abelanger5) [reply frontend/app/src/pages/main/index.tsx:89]: a little off but we should add more tabs to the worker soon anyway

File: frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-metrics.tsx (hunk 7 above)
Comment (abelanger5) [reply frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-metrics.tsx:57]: I think eventually some metrics can take much longer to query, so might be nice to have these be separate

**Pull request merged**